### PR TITLE
feat: dragonfly decorator

### LINKED ISSUE #184: feat: decorate for dragonfly support
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

[Dragonfly](https://www.dragonflydb.io/) aims to be 100% Redis compatible.

It would be great to support it in this plugin.

### Motivation

Dragonfly doesn't provide any additional tool, they suggest to use what the Redis community provides because their APIs aims to be 100% compatible with it.

### Example

```
'use strict'

const Fastify = require('fastify')
const fastifyRedis = require('@fastify/redis')

const fastify = Fastify({ logger: true })

fastify.register(fastifyRedis, { 
  host: '127.0.0.1', 
  password: 'your strong password here',
  port: 6379, // Dragonfly port
  family: 4   // 4 (IPv4) or 6 (IPv6),
  isDragonfly: true
})

fastify.get('/foo', (req, reply) => {
  const { dragonfly } = fastify
  dragonfly.get(req.query.key, (err, val) => {
    reply.send(err || val)
  })
})

fastify.post('/foo', (req, reply) => {
  const { dragonfly } = fastify
  dragonfly.set(req.body.key, req.body.value, (err) => {
    reply.send(err || { status: 'ok' })
  })
})

fastify.listen({ port: 3000 }, err => {
  if (err) throw err
  console.log(`server listening on ${fastify.server.address().port}`)
})
```

### PR DESCRIPTION
**Pull request by ilteoood**
Resolves #184

#### Checklist

- [X] run `npm run test` and `npm run benchmark`
- [X] tests and/or benchmarks are included
- [X] documentation is changed or added
- [X] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -79,6 +79,48 @@ fastify.listen({ port: 3000 }, err => {
 })
 ```
 
+### Accessing the Dragonfly Client
+
+Once you have registered your plugin, you can access the Redis client via `fastify.dragonfly`.
```
Comment (ilteoood) [inline README.md:84]:
```suggestion
Once you have registered your plugin, you can access the Dragonfly client via `fastify.dragonfly`. 
```

**Review by mcollina — CHANGES_REQUESTED** [review]
I don't understand the purpose of this, could you clarify? This seem to only change the name of the decorator.

**Comment by ilteoood** [comment]
Hi @mcollina,
You are right: the purpose of this is just to define a different decorator name if the user wants to use this plugin with Dragonfly instead of Redis.
My idea behind this is just to improve the DX and avoid the user to type fastify.redis when he's not using Redis, but Dragonfly.

**Review by Uzlopak — CHANGES_REQUESTED** [review]
I think, developers will not get psychologycal damage by typing redis instead of dragonfly.

Also we have the namespace option.

So I am -1

**Comment by ilteoood** [comment]
Ok then, at least I tried.
Thank you everyone for your time, feel free to close both the PR and the issue then

**Pull request closed without merging**